File: goBackend/models/WebhookInfo.go
package models

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

type WebhookInfo struct {
	Id              primitive.ObjectID `bson:"_id,omitempty"`
	Name            string             `bson:"name"`
	DestinationUrls []string           `bson:"destinationUrls"`
	Log             []time.Time        `bson:"log"`
	OwnerId         primitive.ObjectID `bson:"ownerId"`
	Active          bool               `bson:"active"`
	Archived        bool               `bson:"boolean"`
	Created         time.Time          `bson:"created"`
}

File: website/models/WebhookInfo.ts
import { ObjectId } from 'mongodb';
import { z } from 'zod';

export type WebhookInfo = {
    _id: ObjectId
    name: string
    destinationUrls: string[]
    ownerId: ObjectId
    active: boolean
    archived: boolean
    created: Date
};

export const webhookInfoSchema = z.object({
    _id: z.optional(z.string()),
    name: z.string().min(1, 'name cannot be empty'),
    destinationUrls: z.array(z.string()),
    ownerId: z.string(),
    active: z.boolean(),
    archived: z.boolean(),
    created: z.coerce.date()
});

export const newWebhookInfoSchema = z.object({
    name: z.string().min(1, 'name cannot be empty'),
    destinationUrls: z.array(z.string()),
});
